service-proxy.js

```diff
@@ -992,6 +1134,16 @@ function reapSinglePeer(hostPort, serviceNames) {
         return;
     }
 
+    if (peer.draining) {
+        // TODO: assuming that it's an ex-affinity drain, would be better to
```

Comment:
my thinking on how to do this so far:
- we really need to reify the drain lifecycle of a peer (maybe also of channel)
- i.e. `peer.draining` should be an object owning the lifecycle, with properties such as
  - `peer.draining.reason` formerly peer.drainReason
  - `peer.draining.timeout` formerly peer.drainTimeout
  - `peer.draining.callback` formerly not available
- then we could branch on say the identity of `peer.draining.callback`
- and we could do such things an `peer.draining.cancel();` or...
- `peer.draining.update(/* new reason, direction, callback, etc, to change the course of a drain while it's still ongoing */);`